src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -588,11 +588,12 @@ public static int getNumberOfChunks(long size) {
         long size = getSize();
         long newSize = LongUtil.intsToLong(endHigh, endLow);
         return network.synchronizer.applyComplexUpdate(owner(), signingPair(),
-                (s, committer) -> overwriteSection(s, committer, fileData,
+                (s, committer) -> clean(s, committer, network, crypto)
+                        .thenCompose(u -> u.left.overwriteSection(u.right, committer, fileData,
                         0L, newSize, network, crypto, monitor)
                         .thenCompose(v -> newSize >= size ? Futures.of(v) :
                                 getUpdated(v, network)
-                                        .thenCompose(f -> f.truncate(v, committer, newSize, network, crypto))))
+                                        .thenCompose(f -> f.truncate(v, committer, newSize, network, crypto)))))
```

Comment:
We can close the scope as soon as we are done with it to reduce indentation hell. here, line 593.

Reply:
re-formatted with a little help from IDE